src/orange/g4org/SolidConverter.cc

```diff
@@ -336,8 +336,55 @@ auto SolidConverter::genericpolycone(arg_type solid_base) -> result_type
 auto SolidConverter::generictrap(arg_type solid_base) -> result_type
 {
     auto const& solid = dynamic_cast<G4GenericTrap const&>(solid_base);
-    CELER_DISCARD(solid);
-    CELER_NOT_IMPLEMENTED("generictrap");
+
+    auto const vtx = solid.GetVertices();
+    CELER_ASSERT(vtx.size() == 8);
+
+    std::vector<GenTrap::Real2> lower(4), upper(4);
+    for (auto i : range(4))
+    {
+        lower[i] = scale_.to<Cone::Real2>(vtx[i].x(), vtx[i].y());
+        upper[i] = scale_.to<Cone::Real2>(vtx[i + 4].x(), vtx[i + 4].y());
+    }
+    real_type hh = scale_(solid.GetZHalfLength());
+
+    return make_shape<GenTrap>(solid, hh, lower, upper);
+}
+
+//---------------------------------------------------------------------------//
+
+//! Convert a trapezoid
+auto SolidConverter::trap(arg_type solid_base) -> result_type
+{
+    auto const& solid = dynamic_cast<G4Trap const&>(solid_base);
+
+    real_type sinphi, cosphi;
+    sincos(solid.GetPhi(), &sinphi, &cosphi);
+    auto tanTheta = std::tan(solid.GetTheta());
+
+    auto hz = scale_(solid.GetZHalfLength());
+    auto hy1 = scale_(solid.GetYHalfLength1());
+    auto hy2 = scale_(solid.GetYHalfLength2());
+    auto hx1 = scale_(solid.GetXHalfLength1());
+    auto hx2 = scale_(solid.GetXHalfLength2());
+    auto hx3 = scale_(solid.GetXHalfLength3());
+    auto hx4 = scale_(solid.GetXHalfLength4());
+    auto dxdzDz = hz * tanTheta * cosphi;
+    auto dydzDz = hz * tanTheta * sinphi;
+    auto dxdyDy1 = scale_(solid.GetTanAlpha1() * solid.GetYHalfLength1());
+    auto dxdyDy2 = scale_(solid.GetTanAlpha2() * solid.GetYHalfLength2());
```

Comment:
I think it's clearer to reuse the locals `hy1` and `hy2` instead of re-fetching it and scalingn it:
```suggestion
    auto dxdyDy1 = y1 * solid.GetTanAlpha1();
    auto dxdyDy2 = y2 * solid.GetTanAlpha2();
```